src/lib/Microsoft.Health.Extensions.Fhir.R4/HttpClientBuilderRegistrationExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Extensions.Host.Auth;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public static class HttpClientBuilderRegistrationExtensions
```

Comment:
>HttpClientBuilderRegistrationExtensions [](http://example.com/codeflow?start=24&length=39)

Is there anything in this class that has a dependency on FHIR R4 libraries?  If not this should be in the Microsoft.Health.Extensions.Fhir project.  By placing it here we will need to duplicate this once R5 or other FHIR versions are released.  If there is a dependency that requires it be separate then this is fine.

Reply:
This class instantiates BearerTokenAuthorizationMessageHandler() which has dependencies on the FhirClientMetric class in Fhir R4 libraries. Because of these existing dependencies, I believe we should keep HttpClientBuilderRegistrationExtensions class separate

However if the above solution is not meeting our needs, a potential idea is to move Fhir.R4.Telemetry folder and HttpClientBuilderRegistrationExtensions class to Microsoft.Health.Extensions.Fhir project now that we've removed FhirOperationException.  Feel free to reopen this thread if you'd like to discuss this further

I moved the ServiceCollectionExtensions class to the Microsoft.Health.Extensions.Fhir project because it has no dependencies on FHIR R4 libraries